src/components/NotificationsDrawer/useNotificationDrawerStore.tsx

```diff
@@ -0,0 +1,33 @@
+import { useContext, useEffect } from 'react';
+import { getModule } from '@scalprum/core';
+import { useSetAtom } from 'jotai';
+import InternalChromeContext from '../../utils/internalChromeContext';
+import { NotificationData, notificationDrawerScopeReadyAtom, notificationDrawerUnreadAtom } from '../../state/atoms/notificationDrawerAtom';
+import { getNotificationsScope } from './NotificationsScope';
+
+export const useNotificationDrawerStore = () => {
+  const { getUserPermissions } = useContext(InternalChromeContext);
+  const setUnreadNotifications = useSetAtom(notificationDrawerUnreadAtom);
+  const setNotificationsDrawerScopeReady = useSetAtom(notificationDrawerScopeReadyAtom);
+
+  async function addScope() {
+    try {
+      const initNotificationScope = await getModule<() => void>('notifications', './initNotificationScope');
+      initNotificationScope();
+      setTimeout(async () => {
+        const { initialize, getState } = getNotificationsScope();
+        getUserPermissions('notifications').then((perms) => {
+          initialize(true, perms).then(() => {
+            setUnreadNotifications(getState().notificationData.some((item: NotificationData) => !item.read));
```

Comment:
^^ As I mentioned above, this state and all that depends on it should be moved to the notifications repository.